chall05/estina.c

```diff
@@ -0,0 +1,47 @@
+/* ************************************************************************** */
+/*                                                                            */
+/*                                                        :::      ::::::::   */
+/*   estina.c                                           :+:      :+:    :+:   */
+/*                                                    +:+ +:+         +:+     */
+/*   By: estina <estina@student.42madrid.com>       +#+  +:+       +#+        */
+/*                                                +#+#+#+#+#+   +#+           */
+/*   Created: 2020/04/16 14:07:38 by estina            #+#    #+#             */
+/*   Updated: 2020/04/16 14:59:57 by estina           ###   ########.fr       */
+/*                                                                            */
+/* ************************************************************************** */
+
+static char	ft_lower(char c)
+{
+	if (c >= 'A' && c <= 'Z')
+		return (c + 32);
+	else
+		return (c);
+}
+
+static int	ft_check_c(char c, int type)
+{
+	if (c == 'c')
+		return (type ? 1 : 0);
+	return (type ? 0 : 1);
+
+}
+
+int			ft_ie_except_after_c(char* str)
+{
+	int		i;
+	int		rule;
+
+	if (!str || !*str)
+		return (0);
+	i = 0;
+	rule = 2;
+	while (str[i] && rule)
+	{
+		if (ft_lower(str[i]) == 'e' && ft_lower(str[i + 1]) == 'i' && i)
```

Comment:
This condition protects you from a crash but keeps you from verificating if a word that starts with `ei` is good or bad.  
In fact you should return `0` if a word starts with `ei` as it would not be prefixed by `c`.  
Test this case on `ei` (or `either`, `eight`... If you want to test it with real words.